### PR TITLE
Add support for websocket cookies

### LINKED ISSUE #9540: Set request and response cookies for websocket connections
We have code that sets up a network connection to a websocket server in `components/net/websocket_loader.rs`. This is initially an HTTP connection, so it should include things like HTTP cookies, but it currently doesn't. We should make the `set_request_cookies` function in `components/net/http_loader.rs` public and call it from `establish_a_websocket_connection`, which should change some test results in `./mach test-wpt tests/wpt/web-platform-tests/websockets`.

@rebstar6 might be interested in working on this, so please don't anybody else start working on this without talking to me first.

### PR DESCRIPTION
**Pull request by rebstar6**
Addresses both cookies in request and response. Resolves #9540.

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9662)

### CODE REVIEW

**Comment by jdm** [comment]
So many passing tests! Delightful!
-S-awaiting-review +S-needs-code-changes 

---

Reviewed 9 of 9 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[components/net/http_loader.rs, line 377 \[r1\]](https://reviewable.io:443/reviews/servo/servo/9662#-KAfMW9RM_E1MeSLBgLB:-KAfMW9RM_E1MeSLBgLC:-913659147) ([raw file](https://github.com/servo/servo/blob/1a6472525f1cebc6138ca99574b34b4fd7f37905/components/net/http_loader.rs#L377)):*
Let's revert this, since we don't use it any more.

---

*[components/net/websocket_loader.rs, line 30 \[r1\]](https://reviewable.io:443/reviews/servo/servo/9662#-KAfMdg2sUM46ChOHerT:-KAfMdg2sUM46ChOHerU:-679033296) ([raw file](https://github.com/servo/servo/blob/1a6472525f1cebc6138ca99574b34b4fd7f37905/components/net/websocket_loader.rs#L30)):*
Let's move this to the next line.

---

*[components/script/dom/websocket.rs, line 492 \[r1\]](https://reviewable.io:443/reviews/servo/servo/9662#-KAfMs1H1vfo_isn5HWK:-KAfMs1H1vfo_isn5HWL:-2089060057) ([raw file](https://github.com/servo/servo/blob/1a6472525f1cebc6138ca99574b34b4fd7f37905/components/script/dom/websocket.rs#L492)):*
There's nothing we can do about an error sending to the resource thread, so ignoring it is fine.

---

*[components/script/dom/websocket.rs, line 504 \[r1\]](https://reviewable.io:443/reviews/servo/servo/9662#-KAfMlruP6O0Ab0Zv_Ag:-KAfMlruP6O0Ab0Zv_Ah:-125416046) ([raw file](https://github.com/servo/servo/blob/1a6472525f1cebc6138ca99574b34b4fd7f37905/components/script/dom/websocket.rs#L504)):*
We might as well inline this in the caller instead of having a separate function for it.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/9662#-:-KAfNOcgP1-DhMhBWrpt:-1242374282)*

**Comment by rebstar6** [comment]
Review status: 6 of 9 files reviewed at latest revision, 4 unresolved discussions.

---

*[components/net/http_loader.rs, line 377 \[r1\]](https://reviewable.io:443/reviews/servo/servo/9662#-KAfMW9RM_E1MeSLBgLB:-KAfQhGd2aJnvuS6zTZp:-499146009) ([raw file](https://github.com/servo/servo/blob/1a6472525f1cebc6138ca99574b34b4fd7f37905/components/net/http_loader.rs#L377)):*
Done.

---

*[components/net/websocket_loader.rs, line 30 \[r1\]](https://reviewable.io:443/reviews/servo/servo/9662#-KAfMdg2sUM46ChOHerT:-KAfU1Cx7waTHuWRSggr:-499146009) ([raw file](https://github.com/servo/servo/blob/1a6472525f1cebc6138ca99574b34b4fd7f37905/components/net/websocket_loader.rs#L30)):*
Done.

---

*[components/script/dom/websocket.rs, line 492 \[r1\]](https://reviewable.io:443/reviews/servo/servo/9662#-KAfMs1H1vfo_isn5HWK:-KAfU8QvrSquq5NqbDLY:-1586489724) ([raw file](https://github.com/servo/servo/blob/1a6472525f1cebc6138ca99574b34b4fd7f37905/components/script/dom/websocket.rs#L492)):*
got it

---

*[components/script/dom/websocket.rs, line 504 \[r1\]](https://reviewable.io:443/reviews/servo/servo/9662#-KAfMlruP6O0Ab0Zv_Ag:-KAfU6Z8KErxMpRenceG:-499146009) ([raw file](https://github.com/servo/servo/blob/1a6472525f1cebc6138ca99574b34b4fd7f37905/components/script/dom/websocket.rs#L504)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/9662)*

**Comment by jdm** [comment]
Looks great! Could you squash these commits into one, too?
-S-awaiting-review +S-needs-squash 

---

Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/9662#-:-KAfVxZs73YtO4FWB3FH:1050758556)*

**Comment by jdm** [comment]
Sorry, looks like test-tidy is complaining:
```
./components/net/websocket_loader.rs:30: trailing whitespace
./components/script/dom/websocket.rs:492: Line is longer than 120 characters
./components/script/dom/websocket.rs:492: missing space after ,
```

File: components/script/dom/websocket.rs [hunk 1]
```diff
@@ -485,6 +486,13 @@ impl Runnable for ConnectionEstablishedTask {
         };
 
         // Step 5: Cookies.
+        if let Some(cookies) = self.headers.get_raw("set-cookie") {
+            for cookie in cookies.iter() {
+                if let Ok(cookie_value) = String::from_utf8(cookie.clone()) {
+                    let _ = ws.global().r().resource_thread().send(SetCookiesForUrl((ws.url).clone(),cookie_value, HTTP));
```
Comment (jdm) [inline components/script/dom/websocket.rs]: This can be `ws.url.clone()` instead of `(ws.url).clone()`, by the way.

**Comment by rebstar6** [comment]
Forgot to rerun after making the previous changes :/ - should be all set now

**Pull request merged**